src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -154,27 +212,26 @@ void Simulator::handle_message(mavlink_message_t *msg) {
 		case MAVLINK_MSG_ID_HIL_SENSOR:
 			mavlink_hil_sensor_t imu;
 			mavlink_msg_hil_sensor_decode(msg, &imu);
-			fill_sensors_from_imu_msg(&_sensor, &imu);
+			update_sensors(&_sensor, &imu);
+			break;
 
-			// publish message
-			if(_sensor_combined_pub == nullptr) {
-				_sensor_combined_pub = orb_advertise(ORB_ID(sensor_combined), &_sensor);
-			} else {
-				orb_publish(ORB_ID(sensor_combined), _sensor_combined_pub, &_sensor);
-			}
+		case MAVLINK_MSG_ID_HIL_GPS:
+			mavlink_hil_gps_t gps_sim;
+			mavlink_msg_hil_gps_decode(msg, &gps_sim);
+			update_gps(&gps_sim);
 			break;
 
-		case MAVLINK_MSG_ID_MANUAL_CONTROL:
+		case MAVLINK_MSG_ID_RC_CHANNELS:
```

Comment:
The manual control message is the right format to do this, as RC channels is an OUTPUT, not an input message. See the MAVLink app for reference. With UDP enabled it might not be necessary in general, so we could also just leave this as-is.